Include/Rover/Select.hpp

```diff
@@ -0,0 +1,304 @@
+#ifndef ROVER_SELECT_HPP
+#define ROVER_SELECT_HPP
+#include <type_traits>
+#include <vector>
+#include "Rover/Autobox.hpp"
+#include "Rover/Evaluator.hpp"
+#include "Rover/Range.hpp"
+
+namespace Rover {
+namespace Details {
+  struct AnyIndex {
+    template<typename T>
+    operator const T() const { return T(); }
```

Comment:
Don't implement this method, it's not necessary and this implementation assumes a default constructor.

Reply:
Yep I added it for debugging, forgot to remove